src/pybamm/solvers/base_solver.py

```diff
@@ -1562,103 +1550,55 @@ def report(string):
                 f"to parameters {model.calculate_sensitivities} using jax"
             )
             jacp = func.get_sensitivities()
-        if use_jacobian:
-            report(f"Calculating jacobian for {name} using jax")
-            jac = func.get_jacobian()
-            jac_action = func.get_jacobian_action()
-        else:
-            jac = None
-            jac_action = None
+        jac = func.get_jacobian() if use_jacobian else None
+        jac_action = func.get_jacobian_action() if use_jacobian else None
 
     elif model.convert_to_format != "casadi":
         y = vars_for_processing["y"]
         jacobian = vars_for_processing["jacobian"]
 
         if model.calculate_sensitivities:
-            raise pybamm.SolverError(  # pragma: no cover
+            raise pybamm.SolverError(
                 "Sensitivies are no longer supported for the python "
                 "evaluator. Please use `convert_to_format = 'casadi'`, or `jax` "
                 "to calculate sensitivities."
             )
-
-        else:
-            jacp = None
-
-        if use_jacobian:
-            report(f"Calculating jacobian for {name}")
-            jac = jacobian.jac(symbol, y)
-            report(f"Converting jacobian for {name} to python")
-            jac = pybamm.EvaluatorPython(jac)
-            # cannot do jacobian action efficiently for now
-            jac_action = None
-        else:
-            jac = None
-            jac_action = None
-
-        report(f"Converting {name} to python")
+        jacp = None
+        jac = jacobian.jac(symbol, y) if use_jacobian else None
+        jac = pybamm.EvaluatorPython(jac) if jac else None
+        jac_action = None
         func = pybamm.EvaluatorPython(symbol)
+        report(f"Converting {name} to python")
 
     else:
         t_casadi = vars_for_processing["t_casadi"]
         y_casadi = vars_for_processing["y_casadi"]
         p_casadi = vars_for_processing["p_casadi"]
         p_casadi_stacked = vars_for_processing["p_casadi_stacked"]
 
-        # Process with CasADi
         report(f"Converting {name} to CasADi")
         casadi_expression = symbol.to_casadi(t_casadi, y_casadi, inputs=p_casadi)
-        # Add sensitivity vectors to the rhs and algebraic equations
         jacp = None
 
         if model.calculate_sensitivities:
-            report(
-                f"Calculating sensitivities for {name} with respect "
-                f"to parameters {model.calculate_sensitivities} using "
-                "CasADi"
+            report(f"Calculating sensitivities for {name} using CasADi")
```

Comment:
its useful to have the printout of `model.calculate_sensitivities` here, so keep this